src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -600,4 +612,16 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
   override def findResumableExecutions(id: WorkflowId)(implicit ec: ExecutionContext): Future[Map[ExecutionDatabaseKey, JobKey]] = {
     globalDataAccess.findResumableJesExecutions(id)
   }
+
+  /**
+    * Will try to instantiate a user authenticated IOInterface if possible.
+    * If not, will default back to the Cromwell authenticated IOInterface.
+    */
```

Comment:
So in the current state you can provide `clientId/Secrets` in the configuration.
If you do, we assume that you want to access GCS as a User for everything except the write of the auth file (which has to be done by cromwell SA). You will then have to provide a bunch of parameters in the workflow options (refresh_token, bucket for the auth file). That's how cromwell is going to run in a firecloud context.
But for GITC there's no user authentication, everything is done with the service account.
This is allows the backend to support both modes, by providing a user authenticated interface if we have everything to do it, and if not then the normal Service Account is used.